### PR TITLE
Introduce DrawingContext Push/PopRenderOptions

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

```
public override void Render(DrawingContext context)
{
    using (context.PushRenderOptions(new RenderOptions { EdgeMode = EdgeMode.Aliased }))
    {
        context.DrawEllipse(....);
    }

    using (context.PushRenderOptions(new RenderOptions { EdgeMode = EdgeMode.Antialias }))
    {
        context.DrawEllipse(....)
    }
}
```
Screenshot 2023-08-31 072045

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

File: src/Avalonia.Base/Media/DrawingContext.cs [hunk 1]
```diff
@@ -400,6 +403,20 @@ public PushedState PushTransform(Matrix matrix)
...
         }
 
+        /// <summary>
+        /// Pushes render options.
+        /// </summary>
+        /// <param name="renderOptions">The render options.</param>
+        /// <returns>A disposable to undo the render options.</returns>
+        public PushedState PushRenderOptions(RenderOptions renderOptions)
+        {
+            PushRenderOptionsCore(renderOptions);
```
Comment (kekekeks) [inline src/Avalonia.Base/Media/DrawingContext.cs:413]: I believe we can implement the merge here.

Comment (Gillibald) [reply src/Avalonia.Base/Media/DrawingContext.cs:413]: The DrawingContext does not know anything about current RenderOptions. Current RenderOptions are stored in the actual implementation.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0038985-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0038988-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Pull request merged**